mlflow/tracking/client.py

```diff
@@ -218,15 +220,19 @@ def log_batch(self, run_id, metrics=(), params=(), tags=()):
 
     def log_artifact(self, run_id, local_path, artifact_path=None):
```

Comment:
We should add tests for the behavior of this method when it's invoked on a directory.